gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/PartitionedRegion.java

```diff
@@ -7164,7 +7167,8 @@ public boolean hasNext() {
           if (this.bucketEntriesIter != null
               && this.bucketEntriesIter.hasNext()) {
             final RegionEntry val = this.bucketEntriesIter.next();
-            if (val != null && !includeHDFSResults() && val.isMarkedForEviction() && !this.forUpdate) {
+            if (val != null && !(val instanceof NonLocalRegionEntry) && !includeHDFSResults()
+                && val.isMarkedForEviction() && !this.forUpdate) {
```

Comment:
Perhaps can just return false from NonLocalRegionEntry.isMarkedForEviction instead of UnsupportedOperationException and avoid the instanceof check here?